docs/dev/framework/image-processing/image-studio.md

```diff
@@ -0,0 +1,287 @@
+---
+title: "Image Studio"
+description: The Image Studio consists of a set of classes and templates that help you generate and output images and metadata.
+aliases:
+  - /framework/image-processing/image-studio/
+---
+
+{{< version "4.10" >}}
+
+The Image Studio consists of a set of classes and templates that help you generate and output images and metadata.
+
+ * [Components](#components)
+ * [Using the FigureBuilder](#using-the-figurebuilder)
+ * [Templating](#templating)
+
+
+## Components
+
+You can find the Image Studio components in the `Contao\CoreBundle\Image\Studio` [namespace][Studio].
+
+
+### Figure
+
+The `Figure` is a data container that groups all data related to an image under one object. Along the responsive
+image with multiple sizes itself, this can also contain image/file metadata (like an alt text or a caption), a secondary
+light box image, image link attributes or additional template options.
+
+The name *figure* originates from the idea that the HTML `<figure>` tag is also a container for an image with metadata.
+The `Figure` object therefore plays well with the default `image.html5` PHP template as well as the Studio's own Twig
+[templates and macros][TwigTemplates] that both render this tag.
+
+{{% notice tip %}}
+Have a look at the [templating section](#templating) for more information on how to output `Figure` data in your
+templates.
+{{% /notice %}}
+
+In most cases the `FigureBuilder` will be the easiest way to create `Figure` objects. You can, however, create them
+yourself as well:
+
+```php
+// Create with explicitly defined arguments
+$figure = new Figure(
+  $studio->createImage('my/image.png', [800, 600, 'crop']),
+  new MetaData([MetaData::VALUE_ALT => 'My Image']),
+  ['data-foo' => 'bar'],
+  $studio->createLightBoxImage('my/image_big.png')
+);
+
+
+// Create with arguments as closures
+$figure = new Figure(
+  $studio->createImage('my/image.png', [800, 600, 'crop']),
+  static function(Figure $figure): ?MetaData {
+      return new MetaData([MetaData::VALUE_ALT => 'My Image']);
+  }
+);
+
+// Trigger the execution of the closure
+$figure->hasMetaData();
+```
```

Comment:
I’m not sure if it is best to have this example here. It is the first source code block on this page and it is code that most of the people never need to write (or never should). Moving the whole figure section down might not be ideal because it makes sense to explain the figure before the figure builder. Maybe just omit this source code example?

Reply:
Or do we have expandable blocks or something?

Reply:
I dropped it in 0f4439cd0d891d2343aa15aa24d945d06a606211. The few people that will need this will figure it out themselves - it's documented in the code as well.